packages/lesswrong/components/tagging/AllWikiTagsPage.tsx

```diff
@@ -341,99 +267,69 @@ const ArbitalRedirectNotice = ({ classes, onDismiss }: {
-
-  // State variable for the current search query
-  const [currentQuery, setCurrentQuery] = useState('');
-  const [showArbitalRedirectNotice, setShowArbitalRedirectNotice] = useState(isArbitalRedirect);
-
-  // Function to handle search state changes
-  const handleSearchStateChange = (searchState: any): void => {
-    setCurrentQuery(searchState.query);
-  };
-
-  const priorityTagIds = filterNonnull(
-    prioritySlugs.map(slug => tags.find(tag => isMatchingCoreTag(tag, slug)))
-  ).map(tag => tag._id);
-
-  const adjustedItems = tags.map(tag => {
-    if (priorityTagIds.includes(tag._id as typeof priorityTagIds[number])) {
-      return {
-        ...tag,
-        parentTagId: null,
-      };
-    }
+  const currentUser = useCurrentUser();
+  const { openDialog } = useDialog();
 
-    return {
-      ...tag,
-      parentTagId: tag.coreTagId ? tag.coreTagId : 'uncategorized-root',
-    };
+  const { data: priorityTagsRaw } = useQuery(tagsBySlugsQuery, {
+    variables: { slugs: prioritySlugs },
+    ssr: true,
   });
 
-  // Add the artificial "Uncategorized" tags to the items list
-  const itemsWithUncategorized = useMemo(() => [
-    ...adjustedItems,
-    uncategorizedRootTag,
-  ], [adjustedItems]);
 
-  // Move the tree-building and sorting logic here
-  const sortedTree = useMemo(() => {
-    const tree = buildTree(itemsWithUncategorized, null, 0);
-
-    return tree.sort((a, b) => {
+  // Only filter and process priority tags once the data is loaded
+  const priorityTags = useMemo(() => {
+    if (!priorityTagsRaw?.TagsBySlugs) return [];
+    const tags = filterNonnull(priorityTagsRaw.TagsBySlugs);
+    if (!tags.length) return [];
+    
+    return [...tags].sort((a: ConceptItemFragment, b: ConceptItemFragment) => {
       const indexA = prioritySlugs.indexOf(a.slug as typeof prioritySlugs[number]);
       const indexB = prioritySlugs.indexOf(b.slug as typeof prioritySlugs[number]);
-
-      // If both items are in the priority list, sort by priority
-      if (indexA !== -1 && indexB !== -1) {
-        return indexA - indexB;
-      }
-      // If only one item is in priority list, it goes first
-      if (indexA !== -1) return -1;
-      if (indexB !== -1) return 1;
-      // Ensure "Uncategorized" is placed at the end
-      if (a.slug === 'uncategorized-root') return 1;
-      if (b.slug === 'uncategorized-root') return -1;
-      // For all other items, sort by baseScore
-      return b.baseScore - a.baseScore;
+      return indexA - indexB;
     });
-  }, [itemsWithUncategorized]);
+  }, [priorityTagsRaw]);
+
+  const [currentQuery, setCurrentQuery] = useState('');
+  const [showArbitalRedirectNotice, setShowArbitalRedirectNotice] = useState(isArbitalRedirect);
 
-  // Component to access search results
-  const CustomStateResults = connectStateResults(({ searchResults }) => {
+  // Function to handle search state changes
+  const handleSearchStateChange = (searchState: any) => {
```

Comment:
Confirming that search does work with the new implementation?

Reply:
Yes, search works